addons/core/functions/fnc_requestRadios.sqf

```diff
@@ -33,62 +33,124 @@ GVAR(VehicleConfigCacheNamespace) setVariable ["TFAR_fnc_requestRadios_lastExec"
+
+        private _allItems = ((getItemCargo (uniformContainer TFAR_currentUnit)) select 0);
+        _allItems append ((getItemCargo (vestContainer TFAR_currentUnit)) select 0);
+        _allItems append ((getItemCargo (backpackContainer TFAR_currentUnit)) select 0);
+
+        {
+            
+            if (_radiosToReplace isEqualTo []) exitWith {};
+
+            if ((_x call TFAR_fnc_isRadio) || {_x == "ItemRadio"}) then {
+                private _index = _radiosToReplace find _x;
+                If (_index >= 0) then {
+                    private _newRadioClass = _response deleteAt _index;
+
+                    _requestedUnit removeItem _x;
+                    If (_requestedUnit canAdd _newRadioClass) then {
+                        _requestedUnit addItem _newRadioClass;
+                    };
+
+                    _radiosToReplace deleteAt _index;
+                    private _settings = _TF_SettingsToCopy deleteAt _index;
+                    if ([_newRadioClass, _settings] call TFAR_fnc_isSameRadio) then {
+                        private _localSettings = TFAR_RadioSettingsNamespace getVariable (format["%1_local", _settings]);
+                        if !(isNil "_localSettings") then {
+                            [_newRadioClass, _localSettings, true] call TFAR_fnc_setSwSettings;
+                        };
+                    };
+
+                    [_newRadioClass, getPlayerUID player, true] call TFAR_fnc_setRadioOwner;
+                };
             };
+            nil
+        } count _allItems;
 
-            {
-                TFAR_currentUnit addItem _x;
-                _copyIndex = [_settingsCount, _copyIndex, _x,_TF_SettingsToCopy] call _fnc_CopySettings;
-                [_x, getPlayerUID player, true] call TFAR_fnc_setRadioOwner;
-            } forEach (if (_linkFirstItem) then {_response select [1]} else {_response});
-        };
-    } else {
-        hintC _response;
-    };
-    call TFAR_fnc_hideHint;
-    //								unit, radios
-    ["OnRadiosReceived", [TFAR_currentUnit, _response]] call TFAR_fnc_fireEventHandlers;
 
-    ["TFAR_RadioRequestResponseEvent", _thisId] call CBA_fnc_removeEventHandler;
-    [[15,"radioRequest",round ((diag_tickTime-TFAR_beta_RadioRequestStart)*1000)]] call TFAR_fnc_betaTracker;//#TODO remove on release
+        call TFAR_fnc_hideHint;
+
+        ["OnRadiosReceived", [TFAR_currentUnit, _response]] call TFAR_fnc_fireEventHandlers;
 
-}] call CBA_fnc_addEventHandlerArgs;
+        ["TFAR_RadioRequestResponseEvent", _thisId] call CBA_fnc_removeEventHandler;
+        [[15,"radioRequest",round ((diag_tickTime-TFAR_beta_RadioRequestStart)*1000)]] call TFAR_fnc_betaTracker;//#TODO remove on release
+    },
+    [_radiosToRequest, _TF_SettingsToCopy, TFAR_currentUnit]
+] call CBA_fnc_addEventHandlerArgs;
 
 [parseText(localize LSTRING(wait_radio)), 10] call TFAR_fnc_showHint;
+
+// after the settings are initialized, we can replace the itemradio with the default class
```

Comment:
This will not be called unless inventory content changed though....
And the server can do that automagically when it sees ItemRadio. And the server knows the default radio settings

Reply:
This was done with the reduce of server computing in mind. So the Client has already evaluated the right radio and just request the id.
If the settings are not synced, the server takes this part.